sqlest/src/main/scala/sqlest/ast/Column.scala

```diff
@@ -119,6 +119,41 @@ sealed trait AliasedColumn[A] extends Column[A] with CellExtractor[ResultSet, A]
 
 }
 
+/**
+ * A column that has an index associated with it.
+ *
+ * This column is only for internal use when accessing generated keys
+ *
+ */
+case class IndexedColumn[A](index: Int)(implicit val columnType: ColumnType[A]) extends Column[A] with CellExtractor[ResultSet, A] {
```

Comment:
You're both correct that it needs to be a CellExtractor that extracts the column by index. However I don't think it needs to be a Column as it is not used within sql statements. It probably should be moved out of this file too